internal/bcdb/db.go

```diff
@@ -60,6 +60,9 @@ type DB interface {
 	// GetTxProof returns intermediate hashes to recalculate merkle tree root from tx hash
 	GetTxProof(userID string, blockNum uint64, txIdx uint64) (*types.GetTxProofResponseEnvelope, error)
 
+	// GetDataProof returns hashes path from value to root in merkle-patricia trie
+	GetDataProof(userID string, blockNum uint64, dbname string, key string, deleted bool) (*types.GetDataProofResponseEnvelope, error)
```

Comment:
Interesting idea. Let me explain original idea - block number used as time marker. So, once user ask for proof for key K inside block N, it doesn't mean that this key was updated in block N, it just mean that at the time of block N creation, value associated with key K was value with hash located at the end of proof.

Reply:
@gennadylaventman I think it should be the version as mentioned by @tock-ibm For example, multiple transactions in a block can blindly write to a KV pair. The end value of the KV pair is the value written by the last transactions. However, the provenance store would return all writes performed to the KV pair. In order to get a proof for any data, the user can pass the block number and the version. 

Now, I wonder whether we take care of all blind writes within a block. I think, we do not. Am I right?

Reply:
We are not, because it will produce multiple trie roots for same block, at least as number of tx in block. It is doable, but will require same change it integration code and will increase block header size. We had discussion with @tock-ibm  and for now, at least, we going with block number. I already have issue to store metadata hash with value, adding this, I think, will solve version problem.

Reply:
How about we allow only one write to a key per block? In general, blind writes are very rare. Even if blind writes are done by the user, if we allow only one blind write to a key per block, we can prove all states present in the provenance store. Having a root hash per transaction would complicate things more.

Reply:
I don't think root hash per tx is related to amount of writes to same key in block because each write to any key updates root hash. If we handle tx as one atomic update, each tx with writes updates root hash. This is why I mentioned root hash per tx. We had discussion with Yoav and we decided to keep current approach, just to store metadata hash with value hash and to make following changes to REST API `ledger/proof/data/db/key?block={blockNum}&deleted={true|false}`

Reply:
Discussed offline. We will make changes in MVCC to allow only one write to a given key per block.